src/gui/attributetable/qgsattributetablemodel.cpp

```diff
@@ -698,17 +698,37 @@ QVariant QgsAttributeTableModel::data( const QModelIndex &index, int role ) cons
   }
 
   QVariant val = mFeat.attribute( fieldId );
+  bool isLink = false;
+  if ( val.type() == QVariant::String )
+  {
+    QString textVal = val.toString();
+    if ( QRegExp( "^(http|https|ftp|file)://\\S+$" ).indexIn( textVal ) != -1 )
```

Comment:
Please use QRegularExpression instead -- QRegExp is deprecated and removed in qt 6. 

Also would be good to move this logic to QgsStringUtils -- there's similar methods in there